### ISSUE TITLE
Enhancement Request - Put passphrase in install pkg

### ISSUE THREAD

**Issue by gdarm** [munkireport/munkireport-php#239]
Since MR 2.4.0.1042 a install pkg will be created for installation instead of a script.
Thats fine. I like it. :)

But I have a passphrase in my config.php.
As soon as I installed new MR 2.4.0.1042 on a client the client could not talk to MR server again because of the missing passphrase.

As a workaround I wrote a nopkg to check if passphrase is missing and add it to the client again, then kill the cfprefsd to ensure the settings are permanent. It's quick and dirty, but it works...

Could you add the passphrase in to MunkiReport.plist file in the package? :)

**Comment by bochoven** [comment]
You could add the passphrase in a postflight script. You may have a point in that the MR installer should not overwrite the passphrase when it's set.

**Comment by gdarm** [comment]
Yes, but thats a workaround too.

While creating the package the script could get the $conf['client_passphrases'] from config.php and write it to the plist MunkiReport.plist


I've found another error.
The MunkiReport.plist will be cached by cfprefsd and will be considered only after a reboot.

**Comment by bochoven** [comment]
I don't think it is a good idea to put the client passphrase in an installer (that can be accessed from the internet). The whole purpose of the client passphrase is to protect your server from unknown/untrusted machines checking in.
You should set the passphrase only when the machine is in a trusted environment.

About the other issue you found, maybe you could file it as a separate issue?

**Comment by bochoven** [comment]
https://github.com/munkireport/munkireport-php/releases/tag/v2.4.2 should not overwrite the passphrase when it is set, could you please test if this is the case?

**Comment by gdarm** [comment]
Sorry for the late reply.

Seems to working fine.
I've added a postinstall_script to kill cfprefsd. Otherwise we have to reboot the client for the changes to take effect on OS X 10.9 and above.
I know there is a better way using a python script. I've done something similar in the past. But currently no time to code it... :(
